$OpenBSD: patch-src_main_c,v 1.17 2014/03/27 10:13:14 dcoppa Exp $

commit 90f6ffbdb7c7cc07cb02b69fcfb3a3844f0180b7
Author: Lars-Dominik Braun <lars@6xq.net>
Date:   Fri Mar 14 11:47:26 2014 +0100

piano: Request track length

The returned length is used as first length estimate before enough
audio data arrives to show the real length and for the eventcmd
API.

commit ebfd7d45c6b8eb5b2a1ee6e97a3f21f5f5bf9b43
Author: Cody P Schafer <devel@codyps.com>
Date:   Thu Mar 13 11:14:21 2014 -0700

fix warnings now noted due to format string checking

--- src/main.c.orig	Sun Sep 15 15:54:04 2013
+++ src/main.c	Thu Mar 27 10:30:08 2014
@@ -268,6 +268,7 @@ static void BarMainStartPlayback (BarApp_t *app, pthre
 		app->player.scale = BarPlayerCalcScale (app->player.gain + app->settings.volume);
 		app->player.audioFormat = app->playlist->audioFormat;
 		app->player.settings = &app->settings;
+		app->player.songDuration = app->playlist->length * 1000;
 		pthread_mutex_init (&app->player.pauseMutex, NULL);
 		pthread_cond_init (&app->player.pauseCond, NULL);
 
@@ -328,7 +329,7 @@ static void BarMainPrintTime (BarApp_t *app) {
 		sign = POSITIVE;
 		songRemaining = -songRemaining;
 	}
-	BarUiMsg (&app->settings, MSG_TIME, "%c%02i:%02i/%02i:%02i\r",
+	BarUiMsg (&app->settings, MSG_TIME, "%c%02i:%02i/%02li:%02li\r",
 			(sign == POSITIVE ? '+' : '-'),
 			songRemaining / 60, songRemaining % 60,
 			app->player.songDuration / BAR_PLAYER_MS_TO_S_FACTOR / 60,
@@ -388,8 +389,7 @@ static void BarMainLoop (BarApp_t *app) {
 		BarMainHandleUserInput (app);
 
 		/* show time */
-		if (app->player.mode >= PLAYER_SAMPLESIZE_INITIALIZED &&
-				app->player.mode < PLAYER_FINISHED_PLAYBACK) {
+		if (app->player.mode < PLAYER_FINISHED_PLAYBACK) {
 			BarMainPrintTime (app);
 		}
 	}
